src/H5Tconv.c

```diff
@@ -281,12 +296,38 @@ H5T__conv_order(const H5T_t *src, const H5T_t *dst, H5T_cdata_t *cdata,
             /* Capability query */
             if (NULL == src || NULL == dst)
                 HGOTO_ERROR(H5E_ARGS, H5E_BADTYPE, FAIL, "not a datatype");
-            if (src->shared->size != dst->shared->size || 0 != src->shared->u.atomic.offset ||
-                0 != dst->shared->u.atomic.offset ||
-                !((H5T_ORDER_BE == src->shared->u.atomic.order &&
-                   H5T_ORDER_LE == dst->shared->u.atomic.order) ||
-                  (H5T_ORDER_LE == src->shared->u.atomic.order &&
-                   H5T_ORDER_BE == dst->shared->u.atomic.order)))
+
+            src_size = src->shared->size;
+            dst_size = dst->shared->size;
+            if (src_size != dst_size)
+                HGOTO_ERROR(H5E_DATATYPE, H5E_UNSUPPORTED, FAIL, "conversion not supported");
+
+            if (src->shared->parent) {
+                if (!H5T_IS_ATOMIC(src->shared->parent->shared))
+                    HGOTO_ERROR(H5E_DATATYPE, H5E_UNSUPPORTED, FAIL, "conversion not supported");
+                src_offset = src->shared->parent->shared->u.atomic.offset;
+                src_order  = src->shared->parent->shared->u.atomic.order;
+            }
+            else {
+                src_offset = src->shared->u.atomic.offset;
+                src_order  = src->shared->u.atomic.order;
+            }
+            if (dst->shared->parent) {
+                if (!H5T_IS_ATOMIC(dst->shared->parent->shared))
+                    HGOTO_ERROR(H5E_DATATYPE, H5E_UNSUPPORTED, FAIL, "conversion not supported");
+                dst_offset = dst->shared->parent->shared->u.atomic.offset;
+                dst_order  = dst->shared->parent->shared->u.atomic.order;
+            }
+            else {
+                dst_offset = dst->shared->u.atomic.offset;
+                dst_order  = dst->shared->u.atomic.order;
+            }
+
+            if (0 != src_offset || 0 != dst_offset)
+                HGOTO_ERROR(H5E_DATATYPE, H5E_UNSUPPORTED, FAIL, "conversion not supported");
+
+            if (!((H5T_ORDER_BE == src_order && H5T_ORDER_LE == dst_order) ||
```

Comment:
Are you disallowing VAX conversions for atomic types?

Reply:
This is the same logic from before, just compacted a bit with temporary variables for readability. For complex numbers, I decided to throw an error for VAX during conversion based off this comment: https://github.com/HDFGroup/hdf5/blob/96acf6c73910b7271bdb2c0248d3b588d9f45e2b/src/H5Tconv_float.c#L302-L306. Looks like that comment got removed when I added the helper function to retrieve the type of floating-point value involved; I'll put that back.

Reply:
Ah, the original code had scrolled off the screen and I missed that this was the existing behavior.